src/rust/src/x509/verify.rs

```diff
@@ -0,0 +1,33 @@
+// This file is dual licensed under the terms of the Apache License, Version
+// 2.0, and the BSD License. See the LICENSE file in the root of this repository
+// for complete details.
+
+use pyo3::IntoPy;
+
+use crate::x509::certificate::Certificate as PyCertificate;
+
+#[pyo3::pyclass(
+    frozen,
+    name = "Store",
+    module = "cryptography.hazmat.bindings._rust.x509"
+)]
+struct PyStore(pyo3::Py<pyo3::types::PyList>);
+
+#[pyo3::pymethods]
+impl PyStore {
+    #[new]
+    fn new<'p>(py: pyo3::Python<'p>, certs: &'p pyo3::types::PyList) -> pyo3::PyResult<Self> {
+        if certs.iter().any(|c| !c.is_instance_of::<PyCertificate>()) {
+            return Err(pyo3::exceptions::PyTypeError::new_err(
+                "cannot initialize store with non-certificate member",
+            ));
+        }
+        Ok(Self(certs.into_py(py)))
```

Comment:
I think we probably want to just extract this into a `Vec`, we want to avoid the risk that the user modifies the list afterwards.

Reply:
Sounds good -- `Vec<Py<PyAny>>` then? I don't think we can fully extract to `Vec<Certificate>` without cloning.

Reply:
`Vec<Py<PyCertificate>>` for now I guess?

Reply:
Done: https://github.com/pyca/cryptography/pull/9411/commits/f088f739c4fcc9c09de66cee34512e29d94dbba1